scheds/rust/scx_lavd/src/bpf/main.bpf.c

```diff
@@ -506,21 +506,21 @@ static struct cpu_ctx *get_cpu_ctx_id(s32 cpu_id)
 
 static struct sys_cpu_util *get_sys_cpu_util_cur(void)
 {
-	if (__sys_cpu_util_idx == 0)
+	if (READ_ONCE(__sys_cpu_util_idx) == 0)
 		return &__sys_cpu_util[0];
 	return &__sys_cpu_util[1];
 }
 
 static struct sys_cpu_util *get_sys_cpu_util_next(void)
 {
-	if (__sys_cpu_util_idx == 0)
+	if (READ_ONCE(__sys_cpu_util_idx) == 0)
 		return &__sys_cpu_util[1];
 	return &__sys_cpu_util[0];
 }
 
 static void flip_sys_cpu_util(void)
 {
-	__sys_cpu_util_idx ^= 0x1;
+	WRITE_ONCE(__sys_cpu_util_idx, __sys_cpu_util_idx ^ 0x1);
```

Comment:
READ_ONCE() should be used for the XOR op.

Reply:
You mean we need to add `READ_ONCE(__sys_cpu_util_idx);` to ensure the load operation of the value of `__sys_cpu_util_idx` is sequenced-before the XOR operation on it right ?

Reply:
Yup

Reply:
So, if you wanna use `READ_ONCE()`, it should look like `WRITE_ONCE(__sys_cpu_util_idx, READ_ONCE(__sys_cpu_util_idx ^ 0x1))`. Adding `READ_ONCE()` as a separate statement doesn't really achieve anything.

However, because the writer is already guaranteed to see the latest value (it's the sole writer and no optimization can break update visibility from oneself), so I don't think `READ_ONCE()` is necessary here. These volatile updates express the relationships between unsynchronized the writer and its readers. Even when there are multiple writers, they're either usually synchronized among themselves or don't really care what other writers are doing.

Reply:
@htejun 
I made the requested changes, thanks for the advise !
By the way, I've removed the `volatile` qualifier , so I think `READ_ONCE()` is still necessary here ?